### PR TITLE
Refactor StateCheck

### LINKED ISSUE #1096: Look at using https://github.com/ronisbr/PrettyTables.jl in code added by #1086
@charleskawczynski suggest looking at https://github.com/ronisbr/PrettyTables.jl to make output for human reading part of #1086 even more beautiful.

### PR DESCRIPTION
**Pull request by charleskawczynski**
### Description

This PR refactors `StateCheck`:
 - Uses fewer `MPIStateArray`/julia internals (e.g., avoiding use of `.parameters`)
 - More general iteration through variables - fully relies on `flattenednames`/`flattened_tup_chain`
 - Uses PrettyTables instead of manually handling formatting.

I would be nice for someone to try this out and make sure that the behavior / format is acceptable. I haven't changed the part of `StateCheck` that prints julia code, so that things are still easily copy-paste-friendly. Also, there is a `TODO`, which seemed a bit odd to me, would be nice to get feedback on that. I like the look of the new tables :slightly_smiling_face:, also, we can easily add row/column coloring, which is kind of nice.

cc: @blallen @christophernhill

Closes #1096, #1095

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
bors try

File: test/Ocean/refvals/test_ocean_gyre_refvals.jl [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-    ["Q", :η, 12, 12, 11, 12],
-    ["Q", :θ, 12, 12, 12, 12],
-    ["s_aux", :y, 12, 12, 12, 12],
-    ["s_aux", :w, 12, 12, 12, 12],
-    ["s_aux", :pkin, 12, 12, 12, 12],
-    ["s_aux", :wz0, 12, 11, 10, 12],
+    ["Q", "η", 12, 12, 11, 12],
+    ["Q", "θ", 12, 12, 12, 12],
+    ["s_aux", "y", 12, 12, 12, 12],
+    ["s_aux", "w", 12, 12, 11, 12],
```
Comment (charleskawczynski) [inline test/Ocean/refvals/test_ocean_gyre_refvals.jl:7]: @blallen I'm not sure why this changed. I'd like to get to the bottom of it, but I'm not sure what's the best way. Any ideas?

Comment (charleskawczynski) [reply test/Ocean/refvals/test_ocean_gyre_refvals.jl:7]: Also, is there a reason that sometimes symbols are used and other times strings?

Comment (charleskawczynski) [reply test/Ocean/refvals/test_ocean_gyre_refvals.jl:7]: Ohhhh, I think I'm seeing the pattern, now..

**Comment by charleskawczynski** [comment]
Didn't realize that symbols were treated differently than strings. This is not exercised in the test suite. Need to make some more changes..

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bump cc: @blallen @christophernhill

**Review by christophernhill — APPROVED** [review]
@charleskawczynski I went through and looked with @jm-c and @glwagner - all looks good to us

File: src/Diagnostics/Debug/StateCheck.jl [hunk 2]
```diff
@@ -332,124 +306,162 @@ function scprintref(cb)
...
+    println(io, "varr = [")
+    for (s1, s2, s3′, s4′, s5′, s6′) in sc.cur_stats_flat
+        s1 = sp[1:(a1l - length(s1))] * "\"" * s1 * "\""
+        s2 = sp[1:(a2l - length(s2))] * "\"" * s2 * "\""
+        s3 = sprintf1(fmt3, s3′)
+        s4 = sprintf1(fmt3, s4′)
+        s5 = sprintf1(fmt3, s5′)
+        s6 = sprintf1(fmt3, s6′)
+        srow = (s3, s4, s5, s6)
+        println(io, " [ ", s1, ", ", s2, join(srow, ","), " ],")
```
Comment (blallen) [inline src/Diagnostics/Debug/StateCheck.jl:353]: I think this line needs another `", ",` after `s2`

Example output after rebasing onto to master: 
```
varr = [
 [ "state",     "ρ"  9.69901051313856066294e-01,  1.00695111308847851106e+00,  1.00000002953972089159e+00,  6.94722590819101589593e-03 ],
 [ "state", "ρu[1]" -4.21771768146742886962e-01,  6.34790159642324547384e-01,  1.59384804664454482470e-01,  2.22844788492525480716e-01 ],
 [ "state", "ρu[2]" -4.37678540756429201863e-01,  4.53959896024816400573e-01, -1.22554654623556756642e-04,  1.73930782418124457722e-01 ],
 [ "state",    "ρθ" -1.13636401173616619076e+00,  1.51145239791767727056e+00,  1.96870293018723595616e-03,  3.82573373360323043535e-01 ],
]
```

should I open up a PR for this one line fix? @charleskawczynski

Comment (charleskawczynski) [reply src/Diagnostics/Debug/StateCheck.jl:353]: Ah, yes, sorry about that. That would be great, thank you!

Comment (charleskawczynski) [reply src/Diagnostics/Debug/StateCheck.jl:353]: I'm surprised I didn't see the diff from this. Are these ref values git-ignored?

**Pull request merged**